katz/src/main/kotlin/katz/Option.kt

```diff
@@ -134,3 +136,23 @@ sealed class Option<out A> {
  * @param default  the default expression.
  */
 fun <B> Option<B>.getOrElse(default: () -> B): B = fold({ default() }, { it })
+
+/**
+ * Companion interface for [Option] type
+ * outside due to https://youtrack.jetbrains.com/issue/KT-10532
+ */
+private interface OptionInstances {
+    fun <A> HK<Option.F, A>.ev(): Option<A> = this as Option<A>
+
+    fun monad() = object : Monad<Option.F> {
+        override fun <A, B> map(fa: HK<Option.F, A>, f: (A) -> B): HK<Option.F, B> =
```

Comment:
Also sometimes you want to provide different instances for `Future<Option>` and `Monad<Option>`in some cases and you don't want to necessarily use inheritance until we have a solid strategy to receive instances in a lang without implicits.